tests/foreman/api/test_errata.py

```diff
@@ -1406,44 +1415,100 @@ def test_errata_installation_with_swidtags(
     """
     module_name = 'kangaroo'
     version = '20180704111719'
-    # setup base_rhel8 and sat_tools repos
-    rhel8_contenthost.create_custom_repos(
-        **{
-            'baseos': settings.repos.rhel8_os.baseos,
-            'appstream': settings.repos.rhel8_os.appstream,
-        }
-    )
-    function_repos_collection_with_manifest.setup_virtual_machine(rhel8_contenthost)
+    org = module_sca_manifest_org
+    lce = errata_host_lce
+    cv = target_sat.api.ContentView(
+        organization=org,
+        environment=[lce],
+    ).create()
 
-    # install older module stream
-    rhel8_contenthost.add_rex_key(satellite=target_sat)
-    _set_prerequisites_for_swid_repos(rhel8_contenthost)
-    _run_remote_command_on_content_host(
-        f'dnf -y module install {module_name}:0:{version}', rhel8_contenthost
+    repos = {
+        'base_os': settings.repos.rhel8_os.baseos,  # base rhel8
+        'sat_tools': settings.repos.rhel8_os.appstream,  # swid prereqs
+        'swid_tags': settings.repos.swid_tag.url,  # module stream pkgs and errata
+    }
+    # associate repos with sat, org, lce, cv, and sync
+    for r in repos:
+        target_sat.cli_factory.setup_org_for_a_custom_repo(
+            {
+                'url': repos[r],
+                'organization-id': org.id,
+                'lifecycle-environment-id': lce.id,
+                'content-view-id': cv.id,
+            },
+        )
+    # promote newest cv version with all repos/content
+    cv = cv_publish_promote(
+        sat=target_sat,
+        org=org,
+        cv=cv,
+        lce=lce,
+    )['content-view']
+    # ak in env, tied to content-view
+    _ak = target_sat.api.ActivationKey(
+        organization=org,
+        environment=lce,
+        content_view=cv,
+    ).create()
+    # register host with ak, succeeds
+    result = rhel_contenthost.register(
+        activation_keys=_ak.name,
```

Comment:
Very close! Just one more that I see.
```suggestion
    ak = target_sat.api.ActivationKey(
        organization=org,
        environment=lce,
        content_view=cv,
    ).create()
    # register host with ak, succeeds
    result = rhel_contenthost.register(
        activation_keys=ak.name,
```